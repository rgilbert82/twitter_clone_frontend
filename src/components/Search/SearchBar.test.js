import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from '.';

describe(SearchBar, () => {
  let component;

  beforeEach(() => {
    component = shallow(<SearchBar />);
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("updates the search field", () => {
    expect(component.state('searchField')).toEqual('');
    component.find('#search_field_input').simulate('change', { target: { value: 'hello' } });
    expect(component.state('searchField')).toEqual('hello');
  });

  it("checks if the form is valid", () => {
    expect(component.instance().isInvalidForm()).toEqual(true);
    component.find('#search_field_input').simulate('change', { target: { value: 'hello' } });
    expect(component.instance().isInvalidForm()).toEqual(false);
  });
});
