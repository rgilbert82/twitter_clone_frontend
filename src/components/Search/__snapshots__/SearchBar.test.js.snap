// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class SearchBar extends _react2.default.Component {
  constructor(props) {
    super(props);

    this.state = { searchField: '' };

    this.isInvalidForm = this.isInvalidForm.bind(this);
    this.updateSearchField = this.updateSearchField.bind(this);
    this.navigateToSearch = this.navigateToSearch.bind(this);
  }

  isInvalidForm() {
    return !this.state.searchField.length > 0;
  }

  updateSearchField(e) {
    this.setState({ searchField: e.target.value });
  }

  navigateToSearch() {
    this.props.redirectToSearch(this.state.searchField);
    this.setState({ searchField: '' });
  }

  render() {
    return (
      _react2.default.createElement('div', { id: 'search_bar', __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this },
        _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this },
          _react2.default.createElement(_reactBootstrap.FormControl, { id: 'search_field_input', type: 'text', onChange: this.updateSearchField, value: this.state.searchField, placeholder: 'Find by hashtag...', __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }),
          _react2.default.createElement(_reactBootstrap.Button, { onClick: this.navigateToSearch, disabled: this.isInvalidForm(), __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }, 'Search'))));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FormGroup
        bsClass="form-group"
>
        <FormControl
                bsClass="form-control"
                componentClass="input"
                id="search_field_input"
                onChange={[Function]}
                placeholder="Find by hashtag..."
                type="text"
                value=""
        />
        <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={true}
                onClick={[Function]}
        >
                Search
        </Button>
</FormGroup>,
      "id": "search_bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "form-group",
        "children": Array [
          <FormControl
            bsClass="form-control"
            componentClass="input"
            id="search_field_input"
            onChange={[Function]}
            placeholder="Find by hashtag..."
            type="text"
            value=""
/>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={true}
            onClick={[Function]}
>
            Search
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control",
            "componentClass": "input",
            "id": "search_field_input",
            "onChange": [Function],
            "placeholder": "Find by hashtag...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Search",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FormGroup
          bsClass="form-group"
>
          <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    id="search_field_input"
                    onChange={[Function]}
                    placeholder="Find by hashtag..."
                    type="text"
                    value=""
          />
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={true}
                    onClick={[Function]}
          >
                    Search
          </Button>
</FormGroup>,
        "id": "search_bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": Array [
            <FormControl
              bsClass="form-control"
              componentClass="input"
              id="search_field_input"
              onChange={[Function]}
              placeholder="Find by hashtag..."
              type="text"
              value=""
/>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={true}
              onClick={[Function]}
>
              Search
</Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "id": "search_field_input",
              "onChange": [Function],
              "placeholder": "Find by hashtag...",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Search",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
